app/controllers/calagator/events_controller.rb

```diff
@@ -20,6 +20,20 @@ def index
                 query.within_dates(@start_date, @end_date) :
                 query.future
 
+   if (@time = params[:time])
+     if (@start_time = Time.zone.parse(@time[:start]) and @end_time = Time.zone.parse(@time[:end]))
```

Comment:
It might be better to use a local variable, like `parsed_start_time` here, to avoid potential confusion caused by setting `@start_time` first to a Time object and then to a string later on.